docs/orleans/grains/grain-references.md

```diff
@@ -0,0 +1,214 @@
+```
+
+From Orleans client code:
+
+```csharp
+Guid playerId = Guid.NewGuid(); // This would typically be read from an HTTP request parameter or elsewhere.
+IPlayerGrain player = client.GetGrain<IPlayerGrain>(playerId);
+```
+
+Grain references contain three pieces of information:
+
+1. The grain _type_, which uniquely identifies the grain class.
+2. The grain _key_, which uniquely identifies a logical instance of that grain class.
+3. The _interface_ which the grain reference must implement.
+
+> [!NOTE]
+> The grain _type_ and _key_ form the [grain identity](grain-identity.md).
+
+Notice that the above calls to <xref:Orleans.IGrainFactory.GetGrain*?displayProperty=nameWithType> accepted only two of those three things:
+
+* The _interface_ implemented by the grain reference, `IPlayerGrain`.
+* The grain _key_, which is the value of `playerId`.
+
+Above, we said that a grain reference contains a grain _type_, _key_, and _interface_ but we have only provided Orleans with the _key_ and _interface_ in our example code. That is because Orleans maintains a mapping between grain interfaces and grain types. When you ask the grain factory for `IShoppingCartGrain`, Orleans consults its mapping to find the corresponding grain type so that it can create the reference. This works when there is only one implementation of a grain interface, but if there are multiple implementations, then you will need to disambiguate them in the `GetGrain` call. For more information, see the next section, [disambiguating grain type resolution](#disambiguating-grain-type-resolution).
+
+> [!NOTE]
+> Orleans generates grain reference implementation types for each grain interface in your application  during compilation. These grain reference implementations inherit from the <xref:Orleans.Runtime.GrainReference?displayProperty=nameWithType> class. `GetGrain` returns instances of the generated <xref:Orleans.Runtime.GrainReference?displayProperty=nameWithType> implementation corresponding to the requested grain interface.
+
+## Disambiguating grain type resolution
+
+When there are multiple implementations of a grain interface, such as in the following example, Orleans attempts to determine the intended implementation when creating a grain reference. Consider the following example, in which there are two implementations of the `ICounterGrain` interface:
+
+```csharp
+public interface ICounterGrain : IGrainWithStringKey
+{
+  ValueTask<int> UpdateCount();
+}
+
+public class UpCounterGrain : ICounterGrain
+{
+  private int _count;
+  public ValueTask<string> UpdateCount() => new(++_count); // Increment count
+}
+
+public class DownCounterGrain : ICounterGrain
+{
+  private int _count;
+  public ValueTask<string> UpdateCount() => new(--_count); // Decrement count
+}
+```
+
+The following call to `GetGrain` will throw an exception because Orleans does not know how to unambiguously map `ICounterGrain` to one of the grain classes.
```

Comment:
```suggestion
The following call to `GetGrain` will throw an exception because Orleans doesn't know how to unambiguously map `ICounterGrain` to one of the grain classes.
```